hera_pspec/pspecdata.py

```diff
@@ -1721,6 +1906,30 @@ def pspec(self, bls1, bls2, dsets, pols, n_dlys=None, input_data_weight='identit
                         if verbose: print("  Computing and multiplying scalar...")
                         pv *= scalar * self.scalar_delay_adjustment(key1, key2, sampling=sampling)
 
+                    #Generate the covariance matrix if error bars provided
+                    if covariance:
+                        if verbose: print(" Building q_hat covariance...")
+                        cov_qv=np.zeros((self.Ntimes,
+                                         self.spw_Ndlys,
+                                         self.spw_Ndlys),dtype=complex)
+                        cov_pv=np.zeros_like(cov_qv)
+                        amat,bmat=np.meshgrid(range(self.spw_Ndlys),
+                                              range(self.spw_Ndlys))
+                        amat=amat.astype(int)
+                        bmat=bmat.astype(int)
+                        for tnum in range(self.Ntimes):
+                            cov_qv[tnum,:,:]=np.vectorize(lambda a,b:\
+                            self.cov_q_hat(a,b,key1,key2,time_indices=[tnum]))\
+                            (amat,bmat)
+                        if verbose: print(" Building p_hat covariance...")
+                        for tnum in range(self.Ntimes):
+                            cov_pv[tnum]=self.cov_p_hat(Mv,cov_qv[tnum,:,:])
+                        if self.primary_beam != None:
+                            cov_pv*=\
+                            (scalar * self.scalar_delay_adjustment(key1, key2,
+                                                                   sampling=sampling))**2.
+                        pol_cov.extend(cov_pv)
```

Comment:
I'm probably misunderstanding something here, but why do we want `extend` rather than `append`? I think `extend` will take each row of the array and append it as its own array, no? And that's presumably not what we want?

Reply:
Extend applies to the zeroth index of an array. We want an array with (nblpairts, ndlys, ndlys) from nblpairs arrays with shape (ntimes,ndlys,ndlys) which is what extend should produce. Using append would result in an array with shape (nblpairs, ntimes, ndlys,ndlys) which is not what we want.

Reply:
Thanks for the clarification!